lib/drivers/medtronic600Driver.js

```diff
@@ -0,0 +1,1221 @@
+
+    getConfigInfo(progress, data, cb) {
+      if (!isBrowser) {
+        data.connect = true;
+        cb(null, data);
+        return;
+      }
+      debug('in getConfigInfo', data);
+
+      let error = null;
+
+      driver.getCnlInfo()
+        .then(() => {
+          debug('BCNL model and serial:', driver.pumpSession.modelAndSerial);
+          data.connect = true;
+        })
+        .then(() => driver.enterRemoteCommandMode())
+        .then(() => driver.togglePassthroughMode(BCNLDriver.PASSTHROUGH_MODE.ENABLE))
+        .then(() => driver.openConnection())
+        .then(() => driver.readPumpInfo())
+        .then(() => driver.getLinkKey())
+        .then(() => driver.negotiateRadioChannel())
+        .then(() => {
+          debug('*** PUMP SESSION', driver.pumpSession);
+        })
+        .then(() => driver.toggleHighSpeedMode(HighSpeedModeCommand.HIGH_SPEED_MODE.ENABLE))
+        .then(() => driver.getPumpTime())
+        .then(() => driver.readBasalPatterns())
+        .then((schedules) => {
+          const settings = {
+            basalSchedules: schedules,
+          };
+          data.settings = _.clone(settings);
+        })
+        .catch((err) => {
+          driver.togglePassthroughMode(BCNLDriver.PASSTHROUGH_MODE.DISABLE)
+            .then(() => driver.exitRemoteCommandMode());
+          error = err;
+        })
+        .then(() => { // Finally
+          progress(100);
+          if (error) {
+            debug('Error getting config info: ', error.message);
+            cb(error.message, null);
+          } else {
+            cb(null, data);
+          }
+        });
+    },
+
+    fetchData(progress, data, cb) {
+      if (!isBrowser) {
+        data.fetchData = true;
+        cb(null, data);
+        return;
+      }
+      debug('in fetchData', data);
+
+      let error = null;
+
+      debug('*** PUMP SESSION', driver.pumpSession);
+      driver.toggleHighSpeedMode(HighSpeedModeCommand.HIGH_SPEED_MODE.DISABLE)
+        .then(() => {
+          debug('Finished getting data');
+        })
+        .catch((err) => {
+          error = err;
+        })
+        // Finally
+        .then(() => driver.togglePassthroughMode(BCNLDriver.PASSTHROUGH_MODE.DISABLE))
+        .then(() => driver.exitRemoteCommandMode())
+        .then(() => {
+          progress(100);
+          if (error) {
+            debug('Error fetching data', error);
+            cb(error.message, null);
+          } else {
+            cb(null, data);
+          }
+        });
+    },
+
+    processData(progress, data, cb) {
+      debug('in processData');
+      cfg.builder.setDefaults({
+        deviceId: driver.pumpSession.modelAndSerial,
+      });
+      progress(100);
+      cb(null, data);
+    },
+
+    uploadData(progress, data, cb) {
+      progress(0);
+
+      const sessionInfo = {
+        deviceTags: ['insulin-pump'],
+        deviceManufacturers: ['Medtronic'],
+        deviceModel: '640g',
+        deviceSerialNumber: '1055866',
```

Comment:
Yes, this is just the proof of concept code. This will be removed in the next round of changes.